lib/http/response/_data_listener.js

```diff
@@ -2,6 +2,6 @@
 
 module.exports = function onData(data) {
   return function appendChunk(chunk) {
-    data.seq = data.seq.push(chunk);
```

Comment:
The immutable API is so confusing. Not knowing that this is and immutable structure initially, I did a number of double takes to figure out why this wasn't causing massive problems.

Reply:
Yep! Personally, I think that's just bad API design to have a `List` behave _similar_ to a JS Array, with similar interface methods, but _different_ return value! 💥